# Chapter 1 - Data Preparation : Pulling and Cleaning the Imaging Dataset

## Summary

This notebook is basically the first step of our project - getting the imaging data and cleaning it up so I can actually use it. I'm going to load up the dataset, take a look at what we're working with, and fix any messy stuff like missing data or weird formatting. Then I'll save a clean version that's ready to go.

We're keeping all the charts and analysis for another notebook so this one doesn't get too messy and it stays focused on just prepping the data.


Pulling the Data
---

Imports 

In [16]:
import pandas as pd

<br><br>
Reading in the Raw Data

In [17]:
df = pd.read_excel('Data/raw_imaging_data.xlsx')

df.head()
df.info()
df.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 405 entries, 0 to 404
Data columns (total 11 columns):
 #   Column             Non-Null Count  Dtype         
---  ------             --------------  -----         
 0   Date of Procedure  405 non-null    datetime64[ns]
 1   Patient ID         405 non-null    int64         
 2   Age                405 non-null    int64         
 3   Gender             405 non-null    object        
 4   Surgical Findings  405 non-null    object        
 5   Surgical Cure      405 non-null    object        
 6   SPECT/CT           230 non-null    object        
 7   Ultrasound         145 non-null    object        
 8   4D CT Scan         228 non-null    object        
 9   Sestamibi          174 non-null    object        
 10  MRI                15 non-null     object        
dtypes: datetime64[ns](1), int64(2), object(8)
memory usage: 34.9+ KB


Date of Procedure     Patient ID        Age 
count                            405     405.000000  405.000000
mean   2020-05-21 18:36:26.666666752  243753.325926   63.007407
min              2012-04-24 00:00:00  150174.000000   16.000000
25%              2017-11-08 00:00:00  209246.000000   56.000000
50%              2021-03-04 00:00:00  251240.000000   65.000000
75%              2023-04-06 00:00:00  281328.000000   71.000000
max              2025-05-13 00:00:00  312172.000000   94.000000
std                              NaN   44989.927258   12.447412

<br><br>
### Exploratory Data Analysis

#### **Dataset Size & Coverage**
- We've got **405 procedures** to work with
- The data spans from **2012 to 2025**
- Most procedures happened around **2021**


#### **Quick Observations**
- No missing data in our main columns (Date of Procedure, Patient ID) 
- Pretty even distribution of patient IDs suggests good coverage across different patients
- The date column is already in a datetime format

Cleaning the Data
---

Quick cleanup steps:
- Check for missing values
- Fix any weird formatting
- Make sure everything looks good before we start analyzing



<br><br>
1. So first we'll comb through and check for any data that might be missing


In [18]:
# Was an error with age column name being wrong so we viwed the columns and realized it was a space after the 'Age'
df.columns



Index(['Date of Procedure', 'Patient ID', 'Age ', 'Gender',
       'Surgical Findings', 'Surgical Cure', 'SPECT/CT', 'Ultrasound',
       '4D CT Scan', 'Sestamibi', 'MRI'],
      dtype='object')

<br><br>
2. Get rid of (if any) whitespaces in Columns

In [19]:
#  Fixed the column names by getting rid of any whitespace after the column names

df.columns = df.columns.str.strip()
print(df.columns)

Index(['Date of Procedure', 'Patient ID', 'Age', 'Gender', 'Surgical Findings',
       'Surgical Cure', 'SPECT/CT', 'Ultrasound', '4D CT Scan', 'Sestamibi',
       'MRI'],
      dtype='object')


<br><br>
3. Check for Columns that have Null values

In [20]:
# Then we checked for null values
print("\n")
print(df.isna().sum())

# Then we found the rows with null ages
print("\n\nRows with null ages:")
print(df[df['Age'].isna()][['Patient ID']])



Date of Procedure      0
Patient ID             0
Age                    0
Gender                 0
Surgical Findings      0
Surgical Cure          0
SPECT/CT             175
Ultrasound           260
4D CT Scan           177
Sestamibi            231
MRI                  390
dtype: int64


Rows with null ages:
Empty DataFrame
Columns: [Patient ID]
Index: []


<br><br>
4. Check if there's any unncesessary columns

In [21]:
# Sometimes Excel exports columns that are unanmed so we'll just get rid of them

# But this gives an error because no unnamed columns so it's just good to check and make sure
# df = df.drop(columns='Unnamed: 0')

## Saving the Cleaned Data

Now that we’ve finished pulling and cleaning the dataset, like fixing the column names, handling missing values, and making sure the structure is correct, we’re saving the final cleaned version to a new Excel file.

We're going to use the cleaned dataset (`Clean_imaging_data.xlsx`) for our later analysis notebooks so we don’t have to redo the cleaning steps again.


In [ ]:
"""
    Save the cleaned DataFrame to an Excel file, 
    This is what we originally had - just saving the cleaned imaging data
    df.to_excel("Data/Clean_imaging_data.xlsx", index=False)
"""

"""
    But then we realized we needed columns for each individual modality that showed 
    whether or not it matched the surgical findings in the exact way we'd need.
    We needed a column that says "yes exact match" for each modality.
    So we changed it into the modality concordance file because the raw imaging 
    data only had what the imaging found - it never compared it to what was 
    actually found in surgery to see if it was a match to what the scan said.
    We had to create columns for that comparison, and that could be done in the 
    modality concordance file, so we commented this out and moved forward with that approach.
"""